### DISCUSSION TITLE
Ticket Verification

### DISCUSSION THREAD

**Discussion by w1lc0x** [StanleyMasinde/events254Web-archive#109] [Ideas]
**Is your feature request related to a problem? Please describe.**
We cannot verify tickets at the moment. We need to come up with a verification system.

**Describe the solution you'd like**
Use any two of the three ways. We can  implement it using barcode ,  qrcode or just a ticket number search.
Have a page where the physical ticket will have the same details as the one retrieved from the database. 
The implementation will be based on search.

**Comment by StanleyMasinde** [comment]
Hi @wilcox98, Thank you for raising this issue/discussion. The Ticket verification feature is already on the Roadmap. The plan is to Use QR code to identify the authenticity of tickets. 

Why QR?
You can write more information on a QR code

Status?
I am currently scouting for the best QR service. [Qrencode](https://fukuchi.org/works/qrencode/) is the latest candidate. It is super fast but does not give room for customizations. A good solution is with little or no dependencies

How urgent is this?
This is not urgent because we are not lanching with payments

**Comment by w1lc0x** [comment]
Hi @StanleyMasinde , I am looking for customizable alternative that also allows live image scanning i.e using a camera  but at the given moment the best contender for me is [qrcode](https://www.npmjs.com/package/qrcodel). We shall have to decide the tool we shall pick on then work with it as most qr engines runs differently and implementation is different

**Comment by StanleyMasinde** [comment]
The link returns a 404

**Comment by w1lc0x** [comment]
Try this one https://www.npmjs.com/package/qrcode

**Comment by StanleyMasinde** [comment]
Looks good but it has 7 dependencies. I'll give it a try

**Comment by w1lc0x** [comment]
I have used it and it should get us out of trouble

**Comment by StanleyMasinde** [comment]
Sure thanks

**Comment by conradKE** [comment]
yoo guys i hope mko fity vitu zangu zitakua poa and ill be back

**Comment by StanleyMasinde** [comment]
Niko poa mkuu. No rush

**Comment by StanleyMasinde** [comment]
> I like this simple ticket design.
![image](https://user-images.githubusercontent.com/35426405/135052416-3d56bb45-cc60-4c50-8d57-5835ba09ad4b.png)

**Comment by StanleyMasinde** [comment]
@wilcox98  I think this API has been stabilised we can use it https://developer.mozilla.org/en-US/docs/Web/API/Barcode_Detection_API

**Comment by StanleyMasinde** [comment]
@wilcox98 I am rewriting the client. 
Please share your ideas of the ticket here so that I can include them in the new version
